src/NorthwindWebsite/Infrastructure/Repositories/Implementation/ProductRepository.cs

```diff
@@ -3,72 +3,73 @@
 using NorthwindWebsite.Entities;
 using NorthwindWebsite.Infrastructure.Repositories.Interfaces;
 
-namespace NorthwindWebsite.Infrastructure.Repositories.Implementation
+namespace NorthwindWebsite.Infrastructure.Repositories.Implementation;
+
+public class ProductRepository : BaseRepository<Product>, IProductRepository
 {
-    public class ProductRepository : BaseRepository<Product>, IProductRepository
+    public ProductRepository(NorthwindContext context)
+        : base(context)
+    {
+    }
+
+    public new async Task<List<Product>> GetAll() =>
+        await _context.Products
+        .AsNoTracking()
+        .AsQueryable()
+        .Include(p => p.Supplier)
+        .Include(p => p.Category)
+        .ToListAsync();
+
+    public async Task<List<Product>> GetLimitedNumberOfProducts(int limit) =>
+        await _context.Products
+        .AsNoTracking()
+        .AsQueryable()
+        .Include(p => p.Supplier)
+        .Include(p => p.Category)
+        .Take(limit)
+        .ToListAsync();
+
+    public async Task<Product> Get(int id, bool skipRelatedItems)
```

Comment:
Regarding the second parameter there is my following idea: let's implement [this](https://github.com/dotnet-fizzyy/Diploma/blob/master/WebAPI/WebAPI/Infrastructure/Postgres/Repository/BaseCrudRepository.cs#L171) method with `params` delegates indicating what kind of properties should be included. It will be automatically inherited in each type of entity repository.